partiql-parser/src/parse/mod.rs

```diff
@@ -27,6 +27,7 @@ use partiql_source_map::metadata::LocationMap;
 #[allow(clippy::unused_unit)]
 #[allow(unused_variables)]
 #[allow(dead_code)]
+#[allow(rust_2018_idioms)]
```

Comment:
Any reason we're choosing to `allow` `rust_2018_idioms` here while in `partiql-parser/src/lib.rs`, we deny it?

Clippy's also giving the following warning:

```
allow(rust_2018_idioms) is ignored unless specified at crate level

warning: allow(rust_2018_idioms) is ignored unless specified at crate level
  --> partiql-parser/src/parse/mod.rs:30:9
   |
30 | #[allow(rust_2018_idioms)]
   |         ^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_attributes)]` on by default
```

Reply:
The `allow` here is for the inclusion of the `LALRPOP` generated code. 

Without it we get: 
```
error: `extern crate` is not idiomatic in the new edition
  --> /Volumes/workplace/partiql-lang-rust/target/debug/build/partiql-parser-12b4184392ff8cc1/out/partiql.rs:15:1
   |
15 | extern crate core;
   | ^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> partiql-parser/src/lib.rs:1:9
   |
1  | #![deny(rust_2018_idioms)]
   |         ^^^^^^^^^^^^^^^^
   = note: `#[deny(unused_extern_crates)]` implied by `#[deny(rust_2018_idioms)]`
help: convert it to a `use`
   |
15 | use core;
   | ~~~

error: `extern crate` is not idiomatic in the new edition
  --> /Volumes/workplace/partiql-lang-rust/target/debug/build/partiql-parser-12b4184392ff8cc1/out/partiql.rs:34:5
   |
34 |     extern crate core;
   |     ^^^^^^^^^^^^^^^^^^
   |
help: convert it to a `use`
   |
34 |     use core;
   |     ~~~

error: outlives requirements can be inferred
    --> /Volumes/workplace/partiql-lang-rust/target/debug/build/partiql-parser-12b4184392ff8cc1/out/partiql.rs:6301:28
     |
6301 |     where Id: IdGenerator, 'input: 'state, Id: 'state
     |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
     = note: `#[deny(explicit_outlives_requirements)]` implied by `#[deny(rust_2018_idioms)]`
help: remove these bounds
     |
6301 -     where Id: IdGenerator, 'input: 'state, Id: 'state
6301 +     where Id: IdGenerator,
     |
```

Reply:
Ah, I see. Guess we'll need the `allow` for now then